// @copilotkit/react-ui/dist/components/index.mjs
import "../chunk-MMVDU6DF.mjs";
import "../chunk-SC6JRFAJ.mjs";
import {
  CopilotSidebar
} from "../chunk-ALIBUJML.mjs";
import "../chunk-WB3YULQ4.mjs";
import {
  CopilotPopup
} from "../chunk-H3EM63WS.mjs";
import "../chunk-KN2GCKBE.mjs";
import "../chunk-C3GSYRC3.mjs";
import "../chunk-GVKA7RQQ.mjs";
import "../chunk-V7W6IM2V.mjs";
import {
  CopilotDevConsole
} from "../chunk-SGFUVPDB.mjs";
import "../chunk-Q5V6S67N.mjs";
import {
  shouldShowDevConsole
} from "../chunk-QN7T3GWI.mjs";
import "../chunk-KXE2JCUH.mjs";
import "../chunk-NRA3CFEE.mjs";
import "../chunk-BH6PCAAL.mjs";
import "../chunk-UFN2VWSR.mjs";
import {
  CopilotChat
} from "../chunk-HKTWKCPS.mjs";
import "../chunk-53CVDVS5.mjs";
import "../chunk-2II3Q27P.mjs";
import {
  RenderImageMessage
} from "../chunk-ULDQXCED.mjs";
import "../chunk-C7OB63U5.mjs";
import "../chunk-32MUWKL3.mjs";
import {
  AssistantMessage
} from "../chunk-L3GZ7TXC.mjs";
import {
  UserMessage
} from "../chunk-HWMFMBJC.mjs";
import {
  Suggestions
} from "../chunk-QGSPTXOV.mjs";
import {
  Suggestion
} from "../chunk-IMBPSLL4.mjs";
import "../chunk-PLHTVHUW.mjs";
import "../chunk-4HUXYD3B.mjs";
import "../chunk-CGEAG65D.mjs";
import "../chunk-QIOJXTIQ.mjs";
import {
  Markdown
} from "../chunk-YTXEWDNC.mjs";
import "../chunk-B3D7U7TJ.mjs";
import {
  useChatContext
} from "../chunk-IEMQ2SQW.mjs";
import "../chunk-KENCH7RN.mjs";
import "../chunk-XWG3L6QC.mjs";
import "../chunk-54JAUBUJ.mjs";
import "../chunk-JGMFJZMG.mjs";
import "../chunk-S5MBUNGN.mjs";
import "../chunk-MRXNTQOX.mjs";
export {
  AssistantMessage,
  CopilotChat,
  CopilotDevConsole,
  CopilotPopup,
  CopilotSidebar,
  Markdown,
  RenderImageMessage,
  Suggestion as RenderSuggestion,
  Suggestions as RenderSuggestionsList,
  UserMessage,
  shouldShowDevConsole,
  useChatContext
};
